### PR TITLE
fix: dependencies of dependencies 

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

fixes: #1198 

Goes with https://github.com/ApeWorX/ape-solidity/pull/86
Requires https://github.com/ApeWorX/ethpm-types/pull/56

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: src/ape/api/projects.py [hunk 1]
```diff
@@ -190,6 +190,13 @@ def version_id(self) -> str:
         Most often, this is either a version number or a branch name.
         """
 
+    @property
+    @abstractmethod
+    def url(self) -> AnyUrl:
```
Comment (fubuloubu) [inline src/ape/api/projects.py:195]: nit, but shouldn't this be `uri`?

File: src/ape/managers/project/dependency.py [hunk 2]
```diff
@@ -77,6 +77,16 @@ def version_id(self) -> str:
         latest_release = github_client.get_release(self.github, "latest")
         return latest_release.tag_name
 
+    @property
+    def url(self) -> AnyUrl:
```
Comment (fubuloubu) [inline src/ape/managers/project/dependency.py:81]: should be `uri`

File: src/ape/api/projects.py (hunk 1 above)
Comment (unparalleled-js) [reply src/ape/api/projects.py]: The type is `AnyUrl`, not `AnyUri`! JK, I am actually sure on the technically correct answer. Is there a difference asking for a friend?

I am happy to switch to `uri` though!

File: src/ape/api/projects.py [hunk 3]
```diff
@@ -422,4 +422,5 @@ def _load_manifest_from_file(file_path: Path) -> Optional[PackageManifest]:
     except ValidationError as err:
         logger.warning(f"Existing manifest file '{file_path}' corrupted. Re-building.")
         logger.debug(str(err))
+        breakpoint()
```
Comment (fubuloubu) [inline src/ape/api/projects.py]:
```suggestion
```

**Review by fubuloubu — APPROVED** [review]
Seems like this impacts some of the testing workflow. May need some additional QA on usage within projects with tests in them to make sure everything still works.

**Comment by unparalleled-js** [comment]
> Seems like this impacts some of the testing workflow. May need some additional QA on usage within projects with tests in them to make sure everything still works.

I tested against like 4 of Yearn repos and I suggest others do the same.

**Comment by fubuloubu** [comment]
> > Seems like this impacts some of the testing workflow. May need some additional QA on usage within projects with tests in them to make sure everything still works.
> 
> I tested against like 4 of Yearn's repos and I suggest others do the same.

Ah, this is good then. Would appreciate one other person doing QA, then I think this can merge

**Review by NotPeopling2day — APPROVED** [review]
Looks good on my end

**Pull request merged**